### DISCUSSION TITLE
Bug: Disabled Models (same model name but different id) Still Appear While Enabled Models Disappear in Model Selection

### DISCUSSION THREAD

**Discussion by honwei189** [open-webui/open-webui#8077] [Issues]
**The problem still exists and has never been solved in 0.5.0.**

**The bug is of critical nature. The utilization of an incorrect model has led to me incurring a different cost.**

When I disable some OpenRouter models (e.g., those with the same name but with a suffix like "(free)" or ":free") under Admin -> Settings > Model, I encounter an issue where the disabled models still appear in the chat screen model selection list when starting a new chat, while the enabled models disappear. (file attached below)

**For example:**

The model list contains two similar entries:
Meta: Llama 3 8B Instruct (Model ID: meta-llama/llama-3-8b-instruct)
Meta: Llama 3 8B Instruct (free) (Model ID: meta-llama/llama-3-8b-instruct:free)

If I disable Meta: Llama 3 8B Instruct, and then open a new chat, the model list incorrectly displays Meta: Llama 3 8B Instruct (the disabled model) while omitting Meta: Llama 3 8B Instruct (free) (the enabled model).

I checked OpenRouter API history log. It's using a disabled model id.

**Model selection list in chat screen:**
![image](https://github.com/user-attachments/assets/1491c10a-769e-4752-bcd8-789ad302bbd1)

**Models list in Admin -> Setting -> Models**
![image](https://github.com/user-attachments/assets/8797b3d6-c0f8-454c-852c-fec052f6a857)

**Comment by tjbck** [comment]
I'd appreciate if you could provide relevant logs and step by step guide on how to reproduce.

Also, please follow the provided issue template.

**Reply by clduab11** [reply]
Hi @tjbck ,

I was going to open a new Discussion around this issue, but it appears as if I'm experiencing the same bug.

In my Models tab (also an Openrouter user), I have approximately 263 models. While the lion's share are disabled (and the enabling/disabling seems to work a LOT better in 0.5.x versions), I have my Models arranged this way...

![image](https://github.com/user-attachments/assets/c5e04c47-b17b-43ac-bb3f-297cdc814d90)

The naming convention is **provider**: _model, parameters, tune (free/local)_

Ideally, this would make it easier to search for Models when going to the chat interface, but while the naming convention stays true even following a cycle of `docker compose down --volumes --remove-orphans`, `docker system prune -af`, `docker compose build`, `docker-compose --profile development up` to launch my stack (can provide .yaml if needed), I see the following when going to New Chat...

![image](https://github.com/user-attachments/assets/fd7e04fb-9082-442a-a261-a044a7b7069d)

Now, what this screenshot shows isn't ORIGINALLY what caused me to look for this issue...looks like OWUI just takes a minute to update the model information when changing the model name to a certain naming convention... but it'd be great if this screenshot above could match the naming convention of the Model section in Admin Settings in a particular _order_ (like alphabetical, which would match my first screenshot in the second screenshot's menu). When I went to first look, all the names were still goofy and searching for the model searched the entire Model section, not just what I had enabled in said Model section.

I hope it wasn't inappropriate to tag you or anything! I'm a pretty new GitHub user and love love love how OWUI changed my everything haha, but figured I could drop some helpful context along. Let me know if there's anything else you'd like to see to help with this!
